test/Succeed/Issue4166.agda

```diff
@@ -0,0 +1,10 @@
+
+open import Agda.Builtin.Bool
+
+open import Issue4166.Import {b = true} as A′
```

Comment:
I seem to not understand the scoping rules any more.
Isn't `open import Issue4166.Import {b = true} as A` the same as
```agda
import Issue4166.Import {b = true} as A
open A
```
(In this case, the instance `i` would be in scope.)
Or, to elaborate further,
```agda
import Issue4166.Import
module A = Issue4166.Import {b = true}
open A
```
So maybe the idea is that `Issue4166.Import.i : {b : Bool} -> Bool` is not longer used in instance search.
Could we have this example in desugared form?  Or add comments how this works so that it is more dummy-friendly?

Reply:
The desugaring is
```agda
import Issue4166.Import as Fresh
module A' = Fresh {b = true}
open A'
```
and the point of the issue is that the contents of `Fresh` should not be in scope.

Reply:
I posted this comment in the issue so that future archaeologists can figure out what's going on.